### PR TITLE
One-time profile set

### LINKED ISSUE #120: [PE-D] Telemarketers can promote themselves to Supervisors
As mentioned in the User Guide, roles can only be Telemarketer or Supervisor.

However, a Telemarketer can promote to a Supervisor at any time just by calling the command, making Supervisor-specific commands (like tag) obsolete as any user can promote themselves to a Supervisor.

Perhaps making the `profile` command a one-time use command would be more appropriate to your application's use case? i.e after setting up your profile, you should not be able to change your profile, and any editing done through the json would delete all the data to avoid misuse.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: atyhamos/ped#10

### LINKED ISSUE #122: [PE-D] Editing and removing of tag by a supervisor role
Though it makes sense that tags can only be edited or removed by supervisors, users can easily switch their profile to supervisor role to make the changes. This restriction will not be useful in this case. Perhaps you can consider removing this restriction entirely? I think it should be fine that anyone can edit the tags.

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: cheweejia/ped#8

### PR DESCRIPTION
**Pull request by gok99**
Fixes #120 
Fixes #122

### CODE REVIEW

File: src/main/java/teletubbies/model/UserPrefs.java [hunk 1]
```diff
@@ -56,8 +61,18 @@ public UserProfile getUserProfile() {
...
 
-    public void setUserProfile(UserProfile userProfile) {
+    @Override
+    public boolean isProfileSet() {
+        return this.isProfileSet;
+    }
+
+    public void setUserProfile(UserProfile userProfile) throws UserRoleSetException {
         requireNonNull(userProfile);
+        System.out.println(isProfileSet);
```
Comment (tanqiya) [inline src/main/java/teletubbies/model/UserPrefs.java:71]: Perhaps remove or log instead

**Review by tanqiya — APPROVED** [review]
LGTM! We should update the UG and perhaps indicate that switching roles requires deleting the `preferences.json` file.

File: src/main/java/teletubbies/model/UserPrefs.java (hunk 1 above)
Comment (gok99) [reply src/main/java/teletubbies/model/UserPrefs.java:71]: haha slipped through again, thanks for the spot

Comment (gok99) [reply src/main/java/teletubbies/model/UserPrefs.java:71]:
```suggestion
```

**Comment by gok99** [comment]
> LGTM! We should update the UG and perhaps indicate that switching roles requires deleting the `preferences.json` file.

Was also wondering if adding this to the UG would make ppl complain that we're just making it harder to cause the issue but not impossible. I think it's fine though, since changing preferences is probably a legitimate use case

**Pull request merged**